app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/order/OrderController.java

```diff
@@ -8,13 +14,57 @@ public class OrderController {
 
     private static final OrderController ORDER_CONTROLLER = new OrderController();
 
+    private Formatter jsonFormatter = ApplicationContext.getImplementation(Formatter.class);
+    private final Logger LOGGER = ApplicationContext.getImplementation(Logger.class);
+
     private OrderController() {}
 
     public static OrderController getInstance() {
         return ORDER_CONTROLLER;
     }
 
-    public String parseOrder(String requestBody) {
-        return "DogCow sent in a lab order";
+    public Order parseOrder(DomainRequest request) {
+        LOGGER.logInfo("Parsing order...");
+        Order order = new Order();
+        var headers = request.getHeaders();
+        var destination = headers.get("Destination");
+        var client = headers.get("Client");
+        var parsedBody = parseBody(request.getBody());
+        order.setDestination(destination);
+        order.setBody(parsedBody);
+        order.setClient(client);
+        return order;
```

Comment:
One thought that could simplify all of this, and so you don't need to manually parse the request body is to do this...

```suggestion
        return formatter.convertToObject(request.getBody(), Order.class);
```

That then means that the HTTP request body would be something like...

```json
{
	"id": "an ID",
	"destination": "Massachusetts",
	"createAt": "2022-12-21T08:34:27Z",
	"client": "MassGeneral",
	"body": "doesn't matter"
}
```

This also means that nothing needs to be passed in the HTTP headers.

Reply:
Switched it to this. Thanks!